src/GovUk.Education.ExploreEducationStatistics.Common/Services/FileTypeService.cs

```diff
@@ -194,27 +89,20 @@ private string GetMimeTypeUsingMimeProject(Stream stream)
             return await stream.GetFileTypeAsync();
         }
 
-        private string GetMimeEncoding(Stream stream)
-        {
-            return GuessMagicInfo(stream, MagicOpenFlags.MAGIC_MIME_ENCODING);
-        }
-
-        private string GuessMagicInfo(Stream fileStream, MagicOpenFlags flag)
+        private string GuessMagicInfo(Stream fileStream,  MagicOpenFlags flag)
         {
-            using var reader = new StreamReader(fileStream);
-
             var dbPath = _configuration.GetValue<string>("MagicFilePath");
             var magic = new Magic(flag, dbPath);
 
-            var bufferSize = reader.BaseStream.Length >= 1024 ? 1024 : (int) reader.BaseStream.Length;
-            return magic.Read(reader.BaseStream, bufferSize);
+            // I have no idea why wrapping the Stream as a StreamReader is necessary, but it is.
+            using var streamReader = new StreamReader(fileStream);
+            return magic.Read(streamReader.BaseStream, 1024); // magic.Read handles if buffer size is < 1024
         }
 
         private bool IsMimeTypeNullOrZip(string mimeType)
         {
-            return mimeType == null ||
-                   mimeType == "application/octet-stream" ||
-                   mimeType == "application/zip";
+            return mimeType == "application/octet-stream"
+                   || mimeType == "application/zip";
```

Comment:
I think I initially removed this because Rider said it was redundant. Looking into it more, `GuessMagicInfo` returns a non-nullable string, and the `Magic` lib we're calling always returns a string and throws an exception otherwise:

```
    public string Read(byte[] buffer, int bufferSize)
    {
      this.ThrowIfDisposed();
      int length = buffer.Length < bufferSize ? buffer.Length : bufferSize;
      return Marshal.PtrToStringAnsi(MagicNative.magic_buffer(this._magic, buffer, length)) ?? throw new MagicException(this.LastError);
    }
```

Since the method is only being called once, I've inlined it and added a comment for clarity.

Reply:
I think you already know, but the original code had a method for detecting zips which included both types - I'm not sure why. And used MimeDetective for zips and left a comment saying it was a better detector for zips - no idea what that was based on. 

I didn't look into why - I just went along with the previous code.